rdkit/Chem/PandasTools.py

```diff
@@ -124,263 +123,300 @@
+
+  # If this point is reached, an error occurred in the previous try block.
+  # Use old patch:
+  return patch_v1()
 
-def is_molecule_image(s):
-    result = False
-    try:
-        # is text valid XML / HTML?
-        xml = minidom.parseString(s)
-        root_node = xml.firstChild
-        # check data-content attribute
-        if root_node.nodeName in ['svg', 'img'] and \
-           'data-content' in root_node.attributes.keys() and \
-           root_node.attributes['data-content'].value == 'rdkit/molecule':
-            result = True
-    except ExpatError:
-        pass  # parsing xml failed and text is not a molecule image
 
-    return result
+def is_molecule_image(s):
+  result = False
+  try:
+    # is text valid XML / HTML?
+    xml = minidom.parseString(s)
+    root_node = xml.firstChild
+    # check data-content attribute
+    if root_node.nodeName in ['svg', 'img'] and \
+       'data-content' in root_node.attributes.keys() and \
+       root_node.attributes['data-content'].value == 'rdkit/molecule':
+      result = True
+  except ExpatError:
+    pass  # parsing xml failed and text is not a molecule image
+
+  return result
 
 
 class RenderMoleculeAdjustment:
-    def __init__(self, inner_adjustment):
-        """Creates a new instance.
+
+  def __init__(self, inner_adjustment):
+    """Creates a new instance.
 
         @param inner_adjustment: The text adjustment that is used if the
             specified text is not valid XML / HTML.
         """
-        self.inner_adjustment = inner_adjustment
+    self.inner_adjustment = inner_adjustment
 
-    def len(self, text):
-        if is_molecule_image(text):
-            return 0
-        else:
-            return self.inner_adjustment.len(text)
+  def len(self, text):
+    if is_molecule_image(text):
+      return 0
+    else:
+      return self.inner_adjustment.len(text)
 
-    def justify(self, texts, max_len, mode='right'):
-        return self.inner_adjustment.justify(texts, max_len, mode)
+  def justify(self, texts, max_len, mode='right'):
+    return self.inner_adjustment.justify(texts, max_len, mode)
 
-    def adjoin(self, space, *lists, **kwargs):
-        return self.inner_adjustment.adjoin(space, *lists, **kwargs)
+  def adjoin(self, space, *lists, **kwargs):
+    return self.inner_adjustment.adjoin(space, *lists, **kwargs)
 
 
 def _get_image(x):
-    """displayhook function for PNG data"""
-    return b64encode(x).decode('ascii')
+  """displayhook function for PNG data"""
+  return b64encode(x).decode('ascii')
 
 
 try:
-    from rdkit.Avalon import pyAvalonTools as pyAvalonTools
-    # Calculate the Avalon fingerprint
+  from rdkit.Avalon import pyAvalonTools as pyAvalonTools
+
+  # Calculate the Avalon fingerprint
 
-    def _fingerprinter(x, y): return pyAvalonTools.GetAvalonFP(
-        x, isQuery=y, bitFlags=pyAvalonTools.avalonSSSBits)
+
+  def _fingerprinter(x, y):
+    return pyAvalonTools.GetAvalonFP(x, isQuery=y, bitFlags=pyAvalonTools.avalonSSSBits)
```

Comment:
Any reason not to just switch to rdkit's pattern fp?

Reply:
Good one for an enhancement PR